src/calendar/config.rs

```diff
@@ -0,0 +1,27 @@
+use crate::clients::google::{DiskStorage, GoogleClient};
+use crate::config::Config;
+
+use super::facade::*;
+use super::google_calendar::Calendar;
+use super::memory_calendar::InMemoryCalendar;
+
+pub struct CalendarConfig {}
+
+impl CalendarConfig {
+    pub fn new() -> Self {
+        Self {}
```

Comment:
to jest absolutnie nie potrzebne możesz przecież robić `CalendarConfig::memory_calendar` jeżeli `memory_calendar` nie przyjmuje `&self` jako argument (a zarówno `memory_calendar` jak i `google_calendar` nie muszą)

Reply:
Ale ogólnie fajny pattern :D

Reply:
W sumie to nawet nie musisz mieć tego w structcie raczej

Reply:
To, że `memory_calendar` nie przyjmuje `&self` to błąd ;)

Generalnie wziąłem ten pattern z Javy. Te `*Config` to są bean factory. W tym structcie znalazłoby się wszystko co tam bym zrobił `@Autowired`. Potrzem konkretne metody przyjmują już konkretne argumenty potrzebne do utworzenia danej implementacji. 

Ale w sumie teraz tak spojrzałem na wszystkie te moje moduły i wszędzie ten struct jest bez dodatkowych danych. Możliwe, że to w Javie jest to jakaś proteza przez to, że część pochodzi z DI a część podaję ręcznie...

Pomyślę o tym jeszcze chwilę ;)

Reply:
Jednak usunąłem te structy. Jak o tym pomyślałem to rzeczywiście wydają mi się niepotrzebne. Najwyżej przyszłość pokaże, że jednak są potrzebne i wtedy refactor ;)